landscape/client/package/tests/test_changer.py

```diff
@@ -165,85 +199,110 @@ def test_unknown_package_id_for_dependency(self):
+            )
+
         return result.addCallback(got_result)
 
     def test_install_unknown_id(self):
-        self.store.add_task("changer",
-                            {"type": "change-packages", "install": [456],
-                             "operation-id": 123})
+        self.store.add_task(
+            "changer",
+            {"type": "change-packages", "install": [456], "operation-id": 123},
+        )
 
         self.changer.handle_tasks()
 
-        self.assertIn("Package data not yet synchronized with server (456)",
-                      self.logfile.getvalue())
+        self.assertIn(
+            "Package data not yet synchronized with server (456)",
+            self.logfile.getvalue(),
+        )
         self.assertTrue(self.store.get_next_task("changer"))
 
     def test_remove_unknown_id(self):
-        self.store.add_task("changer",
-                            {"type": "change-packages", "remove": [456],
-                             "operation-id": 123})
+        self.store.add_task(
+            "changer",
+            {"type": "change-packages", "remove": [456], "operation-id": 123},
+        )
 
         self.changer.handle_tasks()
 
-        self.assertIn("Package data not yet synchronized with server (456)",
-                      self.logfile.getvalue())
+        self.assertIn(
+            "Package data not yet synchronized with server (456)",
+            self.logfile.getvalue(),
+        )
         self.assertTrue(self.store.get_next_task("changer"))
 
     def test_install_unknown_package(self):
         self.store.set_hash_ids({b"hash": 456})
-        self.store.add_task("changer",
-                            {"type": "change-packages", "install": [456],
-                             "operation-id": 123})
+        self.store.add_task(
+            "changer",
+            {"type": "change-packages", "install": [456], "operation-id": 123},
+        )
 
         self.changer.handle_tasks()
 
-        self.assertIn("Package data not yet synchronized with server (%r)" %
-                      b"hash", self.logfile.getvalue())
+        self.assertIn(
+            "Package data not yet synchronized with server (%r)" % b"hash",
+            self.logfile.getvalue(),
+        )
         self.assertTrue(self.store.get_next_task("changer"))
 
     def test_remove_unknown_package(self):
         self.store.set_hash_ids({b"hash": 456})
-        self.store.add_task("changer",
-                            {"type": "change-packages", "remove": [456],
-                             "operation-id": 123})
+        self.store.add_task(
+            "changer",
+            {"type": "change-packages", "remove": [456], "operation-id": 123},
+        )
 
         self.changer.handle_tasks()
 
-        self.assertIn("Package data not yet synchronized with server (%r)" %
-                      b"hash", self.logfile.getvalue())
+        self.assertIn(
+            "Package data not yet synchronized with server (%r)" % b"hash",
```

Comment:
The right conversion would be changing:
`"Package data not yet synchronized with server (%r)" % b"hash"`

to:
`"Package data not yet synchronized with server (b'hash')"`

Notice that if we execute the code, the resultant string would be:
```
>>> "Package data not yet synchronized with server (%r)" % b"hash"
"Package data not yet synchronized with server (b'hash')"
```

I have converted the code to what it should be since there is no reason to use %, format, or f-string. It is just a constant string. **Please make sure this is the expected behavior from the software as it seems to be a bug.**

Reply:
Thanks for changing to a constant string. I can confirm that this is the intended behaviour. The logfile message is indicating what type of data is not yet synchronized with Server (in this case, the `hash`).